### ISSUE TITLE
Add a CreateDeleteRealms scenario to measure realm CRUD performance and concurrency capability

### ISSUE THREAD

**Issue by davoustp** [keycloak/keycloak-benchmark#60]
### Description

Similarly to #56 , a new scenario that creates and delete realms is required to investigate concurrency and also assist in profiling the use case for performance.

This scenario will:

* acquire the service account token
* create a realm with globally unique name and id
* delete the just created realm

### Discussion

https://github.com/keycloak/keycloak/discussions/11074

### Motivation

This new scenario is required to measure the concurrency behavior and performance of Keycloak with a growing number of realms.

**Comment by davoustp** [comment]
@ahus1 created this new issue as decided.

Scenario is quite easy to evolve from #56 , but it turns out that using a service account token does not allow to delete the just created realm, despite adding roles to it (tried `admin` among other things).

Using the main admin account instead of a service account works correctly - but may not be the most appropriate way especially when running concurrently (users generally don't perform several actions with the same account in parallel).

Any suggestion?

BTW, when using the main admin account and having concurrent users running the scenario, I do encounter a consistency issue re: the collection of all user roles (in the infamous `org.keycloak.models.utils.RoleUtils.getDeepUserRoleMappings(UserModel)` method) against stock Keycloak 18.0.0.
I did try to force Quarkus and `agroal` pool to use the `repeatable-read` transaction isolation level by adding the Java prop `-Dquarkus.datasource.jdbc.transaction-isolation-level=repeatable-read`, to no avail (it's actually weird that this transaction isolation level is not explicitly specified in Keycloak).
Checking the issued JDBC calls (using log4jdbc) does not exhibit any obvious problem (changes are all done in a single transaction, so using `repeatable-read` protects multiple-step lookups from seeing changes concurrently performed by other transactions).
So I suspect a consistency issue with the caching layer, but not sure yet.
More on that later, once the scenario is done.

**Comment by ahus1** [comment]
Regarding permissions: I can't name the correct roles from the top of my head. Please submit a draft PR first that I can then use to explore the necessary roles. I would be fine if the draft PR would use admin account - please describe in a comment what would be needed for that to run the scenario.

Consistency with the legacy store might now and then be an issue, I have wondered myself more than once what a good isolation level would be for the old store. Again, when a draft scenario is available, I'll have a look what's happening there. 

As a side note: The new store will use proper optimistic locking. It should be consistent, although it will now and then throw an exception on concurrent changes.

**Comment by davoustp** [comment]
> Regarding permissions: I can't name the correct roles from the top of my head. Please submit a draft PR first that I can then use to explore the necessary roles. I would be fine if the draft PR would use admin account - please describe in a comment what would be needed for that to run the scenario.

Done

> Consistency with the legacy store might now and then be an issue, I have wondered myself more than once what a good isolation level would be for the old store. Again, when a draft scenario is available, I'll have a look what's happening there.

Ok, instruction provided in the PR comments section

> As a side note: The new store will use proper optimistic locking. It should be consistent, although it will now and then throw an exception on concurrent changes.

That's great! Wondering if this optimistic locking implementation embraces multiple step operations...
Will have a look when I get a chance

**Comment by ahus1** [comment]
@davoustp - you've been talking about an inconsistency issue with the legacy store in a concurrent setup. 

I'd be happy to investigate this, please open a new issue with the scenario and the parameters you run, how you setup Keycloak, and what symptoms you see. 

Regarding repeatable read: Well, repeatable read still allows for phantom reads, and once I got the parameters of your setup, the first thing I'll check is that the isolation level is actually set as expected on the JDBC connection.

**Comment by davoustp** [comment]
Should I open the issue against keycloak or keycloak-benchmark?
